### PR TITLE
Gloo Mesh Gateway: N/S extauth docs first pass

### PR DESCRIPTION
**Pull request by kdorosh**
First pass at extauth docs. Tested and implemented flows for ldap, apikey, basic auth, opa. Still need to test and write for oauth, oauth + opa, and custom auth.

### CODE REVIEW

File: docs/content/guides/gateway/auth/extauth/apikey_auth/_index.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+```shell
+curl -v $(BOOKINFO_INGRESS_GATEWAY_URL)/ratings/1 -H "Host: www.example.com"
+```
+
+The above command should return:
+
+```json
+{"id":1,"ratings":{"Reviewer1":5,"Reviewer2":4}}
+```
```
Comment (kdorosh) [inline docs/content/guides/gateway/auth/extauth/apikey_auth/_index.md:62]: admittedly these setup steps are pretty much identical across guides, as i've been standardizing along the way. in the follow up i plan on pulling more of this out into a common file where possible (might not be possible for the oauth guides)

File: docs/content/guides/gateway/auth/extauth/_index.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
+---
+title: External Authentication
+menuTitle: Ext Auth
+weight: 20
+description: Authenticate and authorize requests to your services using Gloo Mesh's external auth service.
+---
+
+{{% notice note %}}
+This section refers specifically to the **Gloo Mesh Enterprise** external auth server.
```
Comment (kdorosh) [inline docs/content/guides/gateway/auth/extauth/_index.md:9]: this is the only section right now, but will be accompanied by custom auth server at the same level in the follow up. hence the nesting here.

File: docs/content/guides/gateway/auth/extauth/apikey_auth/test-apikey-auth-vg.yaml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                      configRef:
+                        name: apikey-auth
+                        namespace: bookinfo
+                  routeAction:
+                    destinations:
+                      - kubeService:
+                          clusterName: mgmt-cluster
+                          name: ratings
+                          namespace: bookinfo
+  deployToIngressGateways:
```
Comment (Sodman) [inline docs/content/guides/gateway/auth/extauth/apikey_auth/test-apikey-auth-vg.yaml:33]: Just as a TODO, we'll need to update this once https://github.com/solo-io/gloo-mesh/pull/1750 merges

File: docs/content/guides/gateway/auth/extauth/passthrough_auth/_index.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+## Passthrough auth vs. Custom Auth vs. Custom Extauth plugin
+You can also implement your own auth with Gloo Mesh with a [Custom Auth server]({{< versioned_link_path fromRoot="/guides/gateway/auth/custom_auth" >}}) or an [Extauth plugin]({{< versioned_link_path fromRoot="/guides/gateway/auth/extauth/plugin_auth" >}}). 
+
+**gRPC Passthrough vs. Custom Auth server**
+With gRPC passthrough, you can leverage other Gloo Mesh extauth implementations (e.g. OIDC, API key, etc.) alongside custom logic. A custom auth server is not integrated with Gloo Mesh extauth so it cannot do this.
+
+**gRPC Passthrough vs. Extauth plugin**
+Using Gloo Mesh to passthrough requests to a separate authentication component eliminates the need to recompile extauth plugins with each version of Gloo Mesh Gateway.
```
Comment (Sodman) [inline docs/content/guides/gateway/auth/extauth/passthrough_auth/_index.md:20]: Did we say we weren't going to use extauth plugins in the mesh context?

File: docs/content/static/content/enterprise_only_feature_disclaimer [hunk 5]
```diff
@@ -0,0 +1 @@
+This feature is available with a **Gloo Mesh Gateway** license only.
```
Comment (Sodman) [inline docs/content/static/content/enterprise_only_feature_disclaimer:1]: Nit: Maybe rename file to `gateway_only_feature_disclaimer` as folks can have enterprise without gateway?

**Review by Sodman — COMMENTED** [review]
LGTM!

File: docs/content/static/content/enterprise_only_feature_disclaimer (hunk 5 above)
Comment (kdorosh) [reply docs/content/static/content/enterprise_only_feature_disclaimer]: https://github.com/solo-io/gloo-mesh/pull/1792/commits/be1c1ed218421051ff27637535ac5791e5ce62be

File: docs/content/guides/gateway/auth/extauth/passthrough_auth/_index.md (hunk 4 above)
Comment (kdorosh) [reply docs/content/guides/gateway/auth/extauth/passthrough_auth/_index.md]: https://github.com/solo-io/gloo-mesh/pull/1792/commits/be1c1ed218421051ff27637535ac5791e5ce62be

**Review by Sodman — APPROVED** [review]
🚀

File: docs/content/guides/gateway/ratelimiting/set/_index.md [hunk 6]
```diff
@@ -43,7 +43,7 @@ Let's run through a simple example that uses set-style rate limiting.
 First, we need to install Gloo Mesh Enterprise (minimum version `1.1`). Please refer to the corresponding
 [installation guide]({{< versioned_link_path fromRoot="/installation/enterprise" >}}) for details.
 
-This guide makes use of the Bookinfo sample application. You can install the application by following the steps in the [Bookinfo deployment section](#bookinfo-deployment).
+This guide makes use of the Bookinfo sample application. You can install the application by following the steps in the [Bookinfo deployment section]({{% versioned_link_path fromRoot="/guides/#bookinfo-deployment" %}}).
```
Comment (npolshakova) [inline docs/content/guides/gateway/ratelimiting/set/_index.md:46]: 🎉

**Pull request merged**